src/app/3d/qgs3dmapcanvasdockwidget.cpp

```diff
@@ -147,6 +158,15 @@ void Qgs3DMapCanvasDockWidget::identify()
   mCanvas->setMapTool( action->isChecked() ? mMapToolIdentify : nullptr );
 }
 
+void Qgs3DMapCanvasDockWidget::toggleNavigationWidget()
+{
+  QAction *action = qobject_cast<QAction *>( sender() );
```

Comment:
You could avoid the sender check here if you make this a slot with a bool argument and connect it directly to the action::toggled signal. It's a bit nicer then retrieving sender() in a slot.